std/algorithm/comparison.d

```diff
@@ -1565,3 +1565,184 @@ unittest
                 2, "two",
                 ));
 }
+
+// TODO Why does this return $(D false) when $(D T) is a $(D string) or array: isImplicitlyConvertible!(bool, T);
+enum implicitlyConvertsToBool(T) = isImplicitlyConvertible!(bool, T) || isArray!T;
+import std.typetuple : allSatisfy;
+
+/** Returns: First Argument (element of $(D a)) whose implicit conversion to
+    $(D bool) is $(D true.
+
+    Similar to behaviour of or operator in dynamic languages such as Lisp's (or
+    a...) and Python's a or ....
+
+    TODO: Is inout Conversion!T the correct return value?
+
+    NOTE: Lazy parameters are currently, too restrictively, inferred by DMD to
+    always throw eventhough they don't need to be. This makes it impossible to
+    mark $(D either) as $(D nothrow). Issue at
+    https://issues.dlang.org/show_bug.cgi?id=12647
+*/
+CommonType!Ts either(Ts...)(lazy Ts a)
```

Comment:
This is a nice function to have. It should restrict `a.length > 1` to avoid goofy calls like `x.either` and replace `implicitlyConvertsToBool` with `ifTestable` or simple use a lambda to check whether a type is testable with `if`.

`lazy` is indeed quite inefficient but I see that as an implementation problem.